nucypher/blockchain/eth/interfaces.py

```diff
@@ -352,8 +325,17 @@ def connect(self):
         # Connect if not connected
         try:
             self.w3 = self.Web3(provider=self._provider)
-            self.tx_machine.w3 = self.w3  # share this web3 instance with the tracker
+            # client mutates w3 instance (configures middleware etc.)
             self.client = EthereumClient(w3=self.w3)
+
+            # web3 instance fully configured; share instance with ATxM and respective strategies
+            speedup_strategy = ExponentialSpeedupStrategy(
+                w3=self.w3,
+                min_time_between_speedups=120,
+            )  # speedup txs if not mined after 2 mins.
+            self.tx_machine = AutomaticTxMachine(
+                w3=self.w3, tx_exec_timeout=self.TIMEOUT, strategies=[speedup_strategy]
+            )
```

Comment:
In terms of code organization, I do find it a bit out-of-place to instantiate these objects here ("connect") but I understand the mechanical reasons.